crates/ark/src/modules/positron/graphics.R

```diff
@@ -1,113 +1,69 @@
 #
 # graphics.R
 #
-# Copyright (C) 2022-2024 by Posit Software, PBC
+# Copyright (C) 2022-2025 by Posit Software, PBC
 #
 #
 
-# Set up plot hooks.
+# Set up "before new page" hooks
 setHook("before.plot.new", action = "replace", function(...) {
-    .ps.Call("ps_graphics_event", "before.plot.new")
+    .ps.Call("ps_graphics_before_new_page", "before.plot.new")
 })
-
 setHook("before.grid.newpage", action = "replace", function(...) {
-    .ps.Call("ps_graphics_event", "before.grid.newpage")
+    .ps.Call("ps_graphics_before_new_page", "before.grid.newpage")
 })
 
-default_device_type <- function() {
-    if (has_aqua()) {
-        "quartz"
-    } else if (has_cairo()) {
-        "cairo"
-    } else if (has_x11()) {
-        "Xlib"
-    } else {
-        stop("This version of R wasn't built with plotting capabilities")
-    }
-}
-
-renderWithPlotDevice <- function(filepath, format, width, height, res, type) {
-    # width and height are in inches and use 72 DPI to create the requested size in pixels
-    dpi <- 72
+# A persistent list mapping plot `id`s to their display list recording.
+# Used for replaying recordings under a new device or new width/height/resolution.
+RECORDINGS <- list()
```

Comment:
We now store `recordPlot()` recordings in a persistent list rather than writing them to disk. This should be a little faster. Once we gain the ability to know when Positron no longer needs a plot, we should also be able to remove recordings from the persistent list. Right now they persist forever.

Reply:
Is it not better to store them in an environment? Or is it because `id` is an integer and not a string?

Reply:
`id` is a string, but remember when you put it in an environment that string is converted to a SYMSXP. @lionel- was worried about the fact that we'd be creating a lot of SYMSXPs that would never get cleaned up. R doesn't ever clean up the SYMSXP list.